src/test/java/hotel/models/RoomTest.java

```diff
@@ -0,0 +1,81 @@
+package hotel.models;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class RoomTest {
+	private Room room1;
+	private Room room2;
+
+	@BeforeEach
+	void setUp() {
+		room1 = new Room(101);
+	}
+
+	@AfterEach
+	void tearDown() {
+		Hotel.getTakenRoomNumbers().clear();
+	}
+
+	/**
+	 * Tests if we can instantiate another room
+	 * with a different number - test passes if
+	 * the newly assigned number is the same as intended
+	 */
+	@Test
+	void setNumberShouldWorkWhenNumberIsUnique() {
+		room2 = new Room(102);
+		assertEquals(102, room2.getNumber());
+	}
+
+	/**
+	 * Test passes if we are not allowed to
+	 * create a room object with the same number
+	 * as the first one
+	 */
+	@Test
+	void setNumberShouldNotWorkWhenNumberIsNotUnique() {
+		room2 = new Room(room1.getNumber());
```

Comment:
There isn't much benefit for room2 to be a property of the test class, since we always initialize it with a different value